.circleci/config.yml

```diff
@@ -0,0 +1,203 @@
+version: 2.1
+
+# Common definition of the build container.
+# Helps keep job configs (below) a bit more concise.
+# circleci/python:* containers are debian based
+py27_container: &py27_container
+    docker:
+        - image: circleci/python:2.7
+    resource_class: small
+    working_directory: ~/project
+py34_container: &py34_container
+    docker:
+        - image: circleci/python:3.4
+    resource_class: small
+    working_directory: ~/project
+py35_container: &py35_container
+    docker:
+        - image: circleci/python:3.5
+    resource_class: small
+    working_directory: ~/project
+py36_container: &py36_container
+    docker:
+        - image: circleci/python:3.6
+    resource_class: small
+    working_directory: ~/project
+py37_container: &py37_container
+    docker:
+        - image: circleci/python:3.7
+    resource_class: small
+    working_directory: ~/project
+py38_container: &py38_container
+    docker:
+        - image: circleci/python:3.8
+    resource_class: small
+    working_directory: ~/project
+
+filter_build_test_package: &filter_build_test_package
+    filters:
+        tags: 
+            ignore: release-*
+        branches:
+            ignore: release-*
+
+# Use this ONLY for release builds.
+filter_release: &filter_release
+    filters:
+        tags:
+            ignore: /.*/
+        branches:
+            only: release-*
```

Comment:
So we'll move to use `release-<version>` branches to designate a new version of the library, along with a manual "approval" step in CircleCI?

Reply:
Exactly. I'm looking at a couple of GitHub and CircleCI features that will "protect" this workflow somewhat. Stuff like, requiring reviews/PR approval, all tests must pass, allowing only certain users to trigger publishing to PyPI, etc. I'm not sure exactly what's doable so it will take a bit of research and experimentation. My goal is to balance speed and convenience with safety and stability. (In the past couple of years, lots of new people have started to use this lib, so protecting the package releases a bit more is important to us.)

Reply:
Awesome! This will be very convenient - thank you for taking the time to put it together. Let me know if there's anything specific you want a hand with!